scripts-base/src/Component.ts

```diff
@@ -0,0 +1,96 @@
+import {
+	ComponentEl,
+	DelegateEvent,
+	DelegateTarget,
+	EventListeners,
+	EventMap,
+	EventMapByElement,
+} from './componentTypes'
+import { matchesSelector } from './utils'
+
+export interface NamedComponent {
+	componentName: string
+}
+
+export type ComponentConstructor<
+	D,
+	E extends ComponentEl,
+	EMap extends EventMap = EventMapByElement<E>
+> = NamedComponent & (new (element: E, data: D) => Component<D, E, EMap>)
+
+export class ComponentInitializationError extends Error {}
+
+export class Component<D = {}, E extends ComponentEl = HTMLElement, EMap extends EventMap = EventMapByElement<E>> {
+	protected readonly getListeners = (): EventListeners<E, EMap> => []
```

Comment:
Popsal bych více formu `E, EMap` i do readme. Zároveň v readme je ukázka dvou případů:
```
['click', this.handleClick],
['click', '.delegateSelector', this.handleDelegateClick],
```
Ale z typového zápisu tady není úplně zřejmé, že se tam může vyskytovat ten delegated selector.

Reply:
V readme to je napsané, tak je uživateli knihovny úplně jedno, co tady je nebo není zřejmé. 🤷‍♂️ Navíc to podle mě zřejmé je ‒ moc jednodušší ten TS kód už podle mě nebude. Nebo máš nějaký konkrétní návrh?